### PR TITLE
Add `GET_FORMAT` Function To OpenSearch SQL Plugin

### PR DESCRIPTION
**Pull request by GabeFernandez310**
Signed-off-by: GabeFernandez310 <Gabriel.Fernandez@improving.com>

### Description
Adds the `get_format` function to the OpenSearch SQL Plugin. It's implementation is aligned with the [MySQL Documentation](https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_get-format).

**Note: This currently only supports a `USA` format for `DATE`, `TIME`, `TIMESTAMP` and `DATETIME`. Support and testing for other formats will be completed in follow-up work**

Examples:

```
opensearchsql> SELECT get_format(DATE, 'USA');
fetched rows / total rows = 1/1
+---------------------------+
| get_format(DATE, 'USA')   |
|---------------------------|
| %m.%d.%Y                  |
+---------------------------+
```

```
opensearchsql> SELECT DATE_FORMAT('2003-10-03',GET_FORMAT(DATE,'USA'));
fetched rows / total rows = 1/1
+----------------------------------------------------+
| DATE_FORMAT('2003-10-03',GET_FORMAT(DATE,'USA'))   |
|----------------------------------------------------|
| 10.03.2003                                         |
+----------------------------------------------------+
```

```
opensearchsql> SELECT DATE_FORMAT('2003-10-03', null);
fetched rows / total rows = 1/1
+-----------------------------------+
| DATE_FORMAT('2003-10-03', null)   |
|-----------------------------------|
| null                              |
+-----------------------------------+
```
 
### Issues Resolved
#722
 
### Check List
- [X] New functionality includes testing.
  - [X] All tests pass, including unit test, integration test and doctest
- [X] New functionality has been documented.
  - [X] New functionality has javadoc added
  - [X] New functionality has user manual doc added
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java [hunk 1]
```diff
@@ -1136,6 +1154,22 @@ private ExprValue exprFromUnixTimeFormat(ExprValue time, ExprValue format) {
...
 
+  /**
+   * get_format implementation for ExprValue.
+   *
+   * @param type ExprValue of the type.
+   * @param format ExprValue of Time/String type
+   * @return ExprValue..
+   */
+  private ExprValue exprGetFormat(ExprValue type, ExprValue format) {
+    String key = type.stringValue() + "-" + format.stringValue();
```
Comment (penghuo) [inline core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java:1165]: Does https://guava.dev/releases/19.0/api/docs/com/google/common/collect/Table.html solve the problem without concat the string value?

File: docs/user/dql/functions.rst [hunk 2]
```diff
@@ -1756,6 +1756,29 @@ Examples::
...
+----------
+
+Description
+>>>>>>>>>>>
+
+Usage: Returns a string value containing string format specifiers based on the input arguments.
+
+Argument type: TYPE, STRING
+TYPE must be one of the following tokens: [DATE, TIME, DATETIME, TIMESTAMP].
+STRING must be one of the following tokens: ["USA", "JIS", "ISO", "EUR", "INTERNAL"] (" can be replaced by ').
```
Comment (penghuo) [inline docs/user/dql/functions.rst:1769]: only "USA" support for now, right?

Comment (GabeFernandez310) [reply docs/user/dql/functions.rst:1769]: Ah yes. I will adjust the documentation for now and update it once the other formats are added.

**Review by Yury-Fridlyand — APPROVED** [review]

File: core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java (hunk 1 above)
Comment (GabeFernandez310) [reply core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java]: Fixed [here](https://github.com/opensearch-project/sql/pull/1299/files#:~:text=//%20Map%20used%20to,.build()%3B), and [here](https://github.com/opensearch-project/sql/pull/1299/files#:~:text=private%20ExprValue%20exprGetFormat,.of()%3B)

**Comment by GabeFernandez310** [comment]
I had to rebase while addressing PR comments. There appears to be a failing test which exists on the `main` branch as well.

File: docs/user/dql/functions.rst (hunk 2 above)
Comment (GabeFernandez310) [reply docs/user/dql/functions.rst]: Fixed [here ](https://github.com/opensearch-project/sql/pull/1299/files#:~:text=os%3E%20select%20GET_FORMAT(DATE%2C%20%27USA%27)%3B)

**Review by dai-chen — COMMENTED** [review]
May I ask where this comes from? Is it part of deprecate legacy engine or other issue? Because I didn't find it in our codebase.

**Comment by GabeFernandez310** [comment]
> May I ask where this comes from? Is it part of deprecate legacy engine or other issue? Because I didn't find it in our codebase.

The `get_format` function was listed along with others in issue #722 to support date and time functions.

**Comment by Yury-Fridlyand** [comment]
It is [one of MySQL functions](https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_get-format) which we're going to support.

**Comment by dai-chen** [comment]
Should we merge IT fix from `main` before merging this?

**Pull request merged**